### PR TITLE
Notify decrypt errors

### PR DESCRIPTION
**Pull request by skywilliamsonl**
Summary: Send a non-fatal error message through the gst pipeline and propagate the error up to the application in a new PlaybackError API.
Type: Feature
Test Plan: Unittests, Component Tests, Fullstack builds.
Jira: RIALTO-463

### CODE REVIEW

**Review by skywojciechowskim — DISMISSED** [review]

File: media/public/include/MediaCommon.h [hunk 1]
```diff
@@ -376,6 +376,15 @@ struct CodecData
     CodecDataType type{CodecDataType::BUFFER}; /**< The codec data type */
 };
 
+/**
+ * @brief None fatel asynchronous errors reported by the player.
```
Comment (aczs) [inline media/public/include/MediaCommon.h:380]: typo

File: proto/mediapipelinemodule.proto [hunk 2]
```diff
@@ -537,6 +537,24 @@ message BufferUnderflowEvent {
     optional int32 source_id = 2 [default = -1];
 }
 
+/**
+ * @brief Event sent by the server when a none fatal error has occurred in the player.
```
Comment (aczs) [inline proto/mediapipelinemodule.proto:541]: non-fatal?

File: media/client/ipc/interface/IMediaPipelineIpcClient.h [hunk 3]
```diff
@@ -95,6 +95,16 @@ class IMediaPipelineIpcClient
      * @param[in] sourceId  : The id of the source that produced the buffer underflow
      */
     virtual void notifyBufferUnderflow(int32_t sourceId) = 0;
+
+    /**
+     * @brief Notifies the client that a none fatal error has occurred in the player.
```
Comment (aczs) [inline media/client/ipc/interface/IMediaPipelineIpcClient.h:100]: non-fatal?

File: media/public/include/IMediaPipelineClient.h [hunk 4]
```diff
@@ -196,6 +196,16 @@ class IMediaPipelineClient
      * @param[in] sourceId  : The id of the source that produced the buffer underflow
      */
     virtual void notifyBufferUnderflow(int32_t sourceId) = 0;
+
+    /**
+     * @brief Notifies the client that a none fatal error has occurred in the player.
```
Comment (aczs) [inline media/public/include/IMediaPipelineClient.h:201]: non-fatal?

File: media/server/gstplayer/interface/IGstGenericPlayerClient.h [hunk 5]
```diff
@@ -149,6 +149,16 @@ class IGstGenericPlayerClient
      * @param[in] mediaSourceType  : The type of the source that produced the buffer underflow
      */
     virtual void notifyBufferUnderflow(MediaSourceType mediaSourceType) = 0;
+
+    /**
+     * @brief Notifies the client that a none fatal error has occurred in the player.
```
Comment (aczs) [inline media/server/gstplayer/interface/IGstGenericPlayerClient.h:154]: non-fatal?

**Review by aczs — APPROVED** [review]

File: media/server/gstplayer/source/tasks/generic/HandleBusMessage.cpp [hunk 6]
```diff
@@ -196,6 +196,72 @@ void HandleBusMessage::execute() const
...
+        }
+        else
+        {
+            RIALTO_SERVER_LOG_WARN("Unknown warning, ignoring %s - %d: %s (%s)",
+                                   GST_OBJECT_NAME(GST_MESSAGE_SRC(m_message)), err->code, err->message, debug);
+        }
+
+        if ((PlaybackError::UNKNOWN != rialtoError) && (m_gstPlayerClient))
+        {
+            const gchar *mimetype = nullptr;
```
Comment (aczs) [inline media/server/gstplayer/source/tasks/generic/HandleBusMessage.cpp:220]: What do you think about giving names to decryptors from which you would be able to get the media type?
For example audio_decryptor (you might also add a number in similar why like we do with appsrc elements)
This means you would have to pass a non-null name in _createDecryptorElement_.
I think that would be less complicated and less error prone code.

**Review by aczs — APPROVED** [review]

File: media/server/gstplayer/source/tasks/generic/HandleBusMessage.cpp (hunk 6 above)
Comment (skywilliamsonl) [reply media/server/gstplayer/source/tasks/generic/HandleBusMessage.cpp:220]: Yeah that makes sense, I think deliver this and ill do that in a followup.

**Review by skywojciechowskim — DISMISSED** [review]

**Review by aczs — APPROVED** [review]

**Review by skywojciechowskim — APPROVED** [review]

**Review by aczs — APPROVED** [review]

**Pull request merged**